File: c/game.c
#include <stdio.h>
#include <ctype.h>
#include "game.h"
#include "constants.h"
#include "strings.h"

const char * handle_input(int input) {
  switch (toupper(input)) {
  case ROCK:
    return STR_ROCK;
  case PAPER:
    return STR_PAPER;
  case SCISSORS:
    return STR_SCISSORS;
  default:
    return STR_INVALID;
  }
}

void print_keys() {
  printf("%s: %i\n", STR_ROCK, ROCK);
  printf("%s: %i\n", STR_PAPER, PAPER);
  printf("%s: %i\n", STR_SCISSORS, SCISSORS);
}

const char *get_win_message(int user_input, int ai_input) {
  if (user_input == ai_input) {
    return STR_TIE;
  } else if (user_input == PAPER && ai_input == ROCK) {
    return STR_USER_WIN;
  } else if (user_input == PAPER && ai_input == SCISSORS) {
    return STR_AI_WIN;
  } else if (user_input == ROCK && ai_input == PAPER) {
    return STR_AI_WIN;
  } else if (user_input == ROCK && ai_input == SCISSORS) {
    return STR_USER_WIN;
  } else if (user_input == SCISSORS && ai_input == PAPER) {
    return STR_USER_WIN;
  } else if (user_input == SCISSORS && ai_input == ROCK) {
    return STR_AI_WIN;
  }
  return STR_INVALID;
}

File: dart/game.dart
import './strings.dart';
import './constants.dart';

void printKeys() {
  print("$STR_ROCK: $KEY_ROCK");
  print("$STR_PAPER: $KEY_PAPER");
  print("$STR_SCISSORS: $KEY_SCISSORS");
}

String handleInput(final int input) {
  switch (input) {
    case KEY_ROCK:
      return STR_ROCK;
    case KEY_PAPER:
      return STR_PAPER;
    case KEY_SCISSORS:
      return STR_SCISSORS;
    default:
      return STR_INVALID;
  }
}

String getWinMessage(final int userInput, final int aiInput) {
  if (userInput == aiInput) {
    return STR_TIE;
  } else if (userInput == KEY_PAPER && aiInput == KEY_ROCK) {
    return STR_USER_WIN;
  } else if (userInput == KEY_PAPER && aiInput == KEY_SCISSORS) {
    return STR_AI_WIN;
  } else if (userInput == KEY_ROCK && aiInput == KEY_PAPER) {
    return STR_AI_WIN;
  } else if (userInput == KEY_ROCK && aiInput == KEY_SCISSORS) {
    return STR_USER_WIN;
  } else if (userInput == KEY_SCISSORS && aiInput == KEY_PAPER) {
    return STR_USER_WIN;
  } else if (userInput == KEY_SCISSORS && aiInput == KEY_ROCK) {
    return STR_AI_WIN;
  } else {
    return STR_INVALID;
  }
}